src/modules/trade/hooks/useSwitchTokensPlaces.ts

```diff
@@ -0,0 +1,43 @@
+import { useCallback } from 'react'
+import { FractionUtils } from 'utils/fractionUtils'
+import { ExtendedTradeRawState } from '../types/TradeRawState'
+import { useIsWrapOrUnwrap } from './useIsWrapOrUnwrap'
+import { useTradeNavigate } from './useTradeNavigate'
+import { useWalletInfo } from 'modules/wallet'
+import { useTradeState } from './useTradeState'
+import { useDerivedTradeState } from './useDerivedTradeState'
+
+export function useSwitchTokensPlaces(stateOverride: Partial<ExtendedTradeRawState> = {}) {
+  const { chainId } = useWalletInfo()
+  const tradeState = useTradeState()
+  const { inputCurrencyAmount, outputCurrencyAmount } = useDerivedTradeState()?.state || {}
+  const isWrapOrUnwrap = useIsWrapOrUnwrap()
+  const tradeNavigate = useTradeNavigate()
+
+  const { inputCurrencyId, outputCurrencyId } = tradeState?.state || {}
+  const updateState = tradeState?.updateState
+
+  return useCallback(() => {
+    if (!inputCurrencyId || !outputCurrencyId || !updateState) return
+    if (isWrapOrUnwrap) return
+
+    updateState({
+      inputCurrencyId: outputCurrencyId,
+      outputCurrencyId: inputCurrencyId,
+      inputCurrencyAmount: FractionUtils.serializeFractionToJSON(outputCurrencyAmount),
+      outputCurrencyAmount: FractionUtils.serializeFractionToJSON(inputCurrencyAmount),
+      ...stateOverride,
+    })
+    tradeNavigate(chainId, { inputCurrencyId: outputCurrencyId, outputCurrencyId: inputCurrencyId })
```

Comment:
One thing to keep in mind is that if we use this for Swap, there's special handling logic to disable BUY orders for ETH flow in the reducer

Check the usage of this fn https://github.com/cowprotocol/cowswap/blob/628c62596d65e0761ccf70677a55bec9a0a36411/src/legacy/state/swap/reducer.ts#L143

Reply:
Thank you! We don't use it for Swap, but we should add TODO at lest for it. Thanks!
@nenadV91 you can include a TODO comment after you merge this